app/javascript/controllers/ui/filter_controller.js

```diff
@@ -2,18 +2,23 @@ import { Controller } from "@hotwired/stimulus";
 
 export default class UIFilter extends Controller {
   static targets = ["source", "item"];
+  static values = {
+    mode: String,
+  };
 
   connect() {}
 
   filter(event) {
     let lowerCaseFilterTerm = this.sourceTarget.value.toLowerCase();
-    let regex = new RegExp("^" + lowerCaseFilterTerm);
```

Comment:
I have removed this since typing a "." at the search bar it was returning all the results, since . in regexp matches any character excepting line breaks.

So i replaced this with includes and startsWith.

Reply:
I think regex is a superset way more powerful and flexible than this 2 modes. 
I think we can consider the user of this lib as an intelligent and knowledgeable being and instead of provide a way to define the regex that the matching will use so that they can compose a "start with" or "contains" or "ends_with" or "what-have-you" pattern.
Eg:

```suggestion
const pattern = this.patternValue.replace("#{input}", lowerCaseFilterTerm)
let regex = new RegExp(pattern)
```

Reply:
I propose to use the modes to allows the user to select within `plain-text` and `regex` so the user is aware of what happen if he uses a `.` (only if that user knows about regex of course)

Reply:
I like your idea, maybe to avoid the . Problem we could replace the dots with \. ? That would work I think. But the warning idea is not bad.